src/jitlayers.cpp

```diff
@@ -1048,13 +1048,19 @@ static void handleAllErrors(ArchiveWriterError E, HandlerT Handler) {
         Handler(E);
 }
 #endif
+#if JL_LLVM_VERSION >= 60000
+static void reportWriterError(const ErrorInfoBase &E) {
+    std::string err = E.message();
+    jl_safe_printf("ERROR: failed to emit output file %s\n", err.c_str());
+#else
 static void reportWriterError(ArchiveWriterError E) {
     std::string ContextStr = E.first.str();
     std::string err;
     if (!ContextStr.empty())
         err += ContextStr + ": ";
     err += E.second.message();
     jl_safe_printf("ERROR: failed to emit output file %s\n", err.c_str());
+#endif
```

Comment:
You could avoid having to write out the `jl_safe_printf` twice by moving this `#endif` above the call here

Reply:
I would rather move the `}` into the conditional branch since the function implementations are completely. different.

Reply:
ok